### ISSUE TITLE
Question: UniNet vs. UNetDS2GN - what is used in MVSnet / R-MVSnet papers? 

### ISSUE THREAD

**Issue by imagelina** [YoYo000/MVSNet#44]
HI, @YoYo000 !

I see that in the model.py in all the inference routines you actually employ 2D Unet (UNetDS2GN) instead of relatively simple 8-layer UniNet and that change is not mentioned in either of your papers. 

As I understand this change might have noticable impact on the performance as the UNetDS2GN performs multi-scale processing and should be able encode better the fine feature details while UniNet does not.

Was this change implemented after R-MVSnet paper? Is it a a new iterative improvement of the model?

**Comment by YoYo000** [comment]
In R-MVSNet paper I use the simple UniNet. 

I change to UNetDS2GN later as it performs slightly better in **depthmap** reconstruction. But notice that they produce **final point clouds** with almost the same quality.

**Comment by imagelina** [comment]
I see. Thanks for you reply.

It is interesting that despite of the performance boost in estimation of individual depth maps due to modification of the model's architecture, the reconstruction quality of final point clouds is not affected....

Does this mean that the fusion and refinement strategies you employ somewhat inappropriate as they neglect the full potential of the model's output? 

it's like if a sport race car is fully equipped to win the race but the pre-start inspection engineers suddenly decide to remove the most valuable parts from it an leave the driver to deal with less-advantageous vehicle. 

Have you experimented more on this subject?

**Comment by YoYo000** [comment]
@imagelina I think the post-processing steps are correctly implemented. The difference between depth map evaluation and point cloud evaluation is that point clouds are fused from **multi-view** depth maps, which is rather redundant than a single view depth map. 

You can imagine object A is visible in both image 0 and 1. We successfully reconstruct A in depth map 0 but fail in depth map 1, however, it will not affect the final point cloud evaluation.

This is also part of the reason why 3D CNN regularization is better than GRU but R-MVSNet produce similar result to MVSNet.

**Comment by tatsy** [comment]
Hi @YoYo000,
I see that the difference of UniNet and UNet does not have much impact for the final point cloud, but I am wondering if it could affect the required training time until convergence, because the number of parameters is much larger in UNet than those in UniNet.

In the paper, you mentioned you trained by 100,000 iterations, and according to the above discussion, you used UniNet for that. Is the training by 100,000 iterations also enough for UNet?

Honestly, I am now trying to reproduce your result using my own implementation in PyTorch. However, my result is rather worse than yours and now I'm checking the performance using UNet and train it by 100,000 iterations.

Thank you very much for your help.

**Comment by YoYo000** [comment]
@tatsy In MVSNet/R-MVSNet papers I was using UNet. In BlendedMVS I changed to UniNet. 

However, I did not found notable differences between these two networks in both performance and convergence. 100,000 iterations are enough for both networks.

**Comment by tatsy** [comment]
@YoYo000 Alright, it's very important information for me. As far as I tested, UNet is slightly better than the UniNet, but I keep on trying to revise my code. Thank you very much for your kind reply.

**Comment by YoYo000** [comment]
> @YoYo000 Alright, it's very important information for me. As far as I tested, UNet is slightly better than the UniNet, but I keep on trying to revise my code. Thank you very much for your kind reply.

Ops... Sorry it is UniNet (the shallow 5-layer one) for MVSNet/R-MVSNet, UNet for BlendedMVS. Sorry for my mistake - -

**Comment by tatsy** [comment]
@YoYo000 Oh, I see that you used UniNet for the paper results. So, the currently distributed weight files in README.md are not the ones used in the papers, right?

When I tested the MVSNet in my environment, the accuracy was slightly different from the numbers in the paper.
